### PR TITLE
Parallelise caching, with updated progressbar

### PR DESCRIPTION
**Pull request by martindurant**
@mmccarty - not yet right, but a start

### CODE REVIEW

**Comment by martindurant** [comment]
out

File: intake/source/cache.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ def sanitize_path(path):
     return path
 
 
+display = {}
```
Comment (mmccarty) [inline intake/source/cache.py]: Why is ``display`` a dict key with ints instead of an array?

Comment (martindurant) [reply intake/source/cache.py]: Should probably be a set? I was using it differently before, but the values are no longer useful.

**Comment by martindurant** [comment]
out

File: intake/source/cache.py (hunk 1 above)
Comment (mmccarty) [reply intake/source/cache.py]: Yeah, based on the context it which it is used, I think a set would work better.

**Comment by mmccarty** [comment]
Looks good!  Ready to merge?

**Comment by martindurant** [comment]
Do you think the display parameter should be in the config and/or environment parameter and/or cache spec?

**Comment by mmccarty** [comment]
You could add it to the config and env.  I wouldn't put it in the cache spec.

**Pull request merged**